scripts/kmod_build.py

```diff
@@ -147,12 +147,16 @@ def build_module(output_dir, module_ko, kdir, module_dir, make_args, extra_cflag
 
     search_path.extend([str.format(d, kdir=abs_kdir, arch=arch) for d in kernel_search_paths])
     kconfig = os.path.join(abs_kdir, "linux", "kconfig.h")
-    for src_rel in args.sources:
-        if args.common_root:
-            dest = os.path.join(output_dir, os.path.relpath(src_rel, args.common_root))
-        else:
-            dest = os.path.join(output_dir, src_rel)
-        deps.extend(copy_with_deps.copy_with_deps(src_rel, dest, search_path, [kconfig]))
+    root = args.common_root
+    for src in args.sources:
+        if os.path.isabs(src):
```

Comment:
I wonder if this check isn't actually correct - if `srcDir` is a relative path, we won't call `os.path.relpath`, and might actually get incorrect destination paths.

We need to ensure that every source file is under the common root, *even if the common root is relative*.

So possibly this should be more like:
```python
for src in args.sources:
    src = os.path.relpath(src, root)
    if src.startswith("../"):
        # Error
    dest = ...
    deps.extend(...)
```
Although I'm not sure of the most robust mechanism for ensuring that the sources are all subdirectories of the root - this thread has a few different suggestions: https://stackoverflow.com/questions/3812849/how-to-check-whether-a-directory-is-a-sub-directory-of-another-directory

Reply:
Python 3.5 implements `commonpath` [method](https://docs.python.org/3.7/library/os.path.html#os.path.commonpath) but, so far we need to support Python 2.7 also, thus suggested change is the right one. Changed